### PR TITLE
Fix curved mesh support for FD

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TimeDerivative.hpp [hunk 1]
```diff
@@ -79,18 +79,14 @@ struct TimeDerivative {
             .is_identity(),
         "Moving mesh is only partly implemented in ValenciaDivClean. If you "
         "need this look at the complete implementation in GhValenciaDivClean. "
-        "You will at least need to get the normal vectors correct, multiply "
-        "the boundary correction by the determinant of the Jacobian on the "
-        "faces and use det_cell_centered_inv_jacobian in "
-        "add_cartesian_flux_divergence, but there might be more that hasn't "
-        "been updated.");
+        "You will at least update the high-order boundary correction code to "
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TimeDerivative.hpp]: need to

Comment (kidder) [reply src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TimeDerivative.hpp]: is the assumption on line 61 still being made?

File: src/Evolution/DgSubcell/Actions/ReconstructionCommunication.hpp [hunk 2]
```diff
@@ -205,26 +205,32 @@ struct SendDataForReconstruction {
...
-                           orientation);
-        } else {
-          std::copy(sliced_data_in_direction.begin(),
-                    sliced_data_in_direction.end(),
-                    subcell_data_to_send.begin());
-        }
+        // Note: Currently we interpolate our solution to our neighbor FD grid
+        // even when grid points align but are oriented differently. There's a
+        // possible optimization for the rare (almost never?) edge case where
+        // two blocks have the same may but different orientation. Since this
```
Comment (kidder) [inline src/Evolution/DgSubcell/Actions/ReconstructionCommunication.hpp]: I don't understand this line of the comment...

File: tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/Test_TimeDerivative.cpp [hunk 3]
```diff
@@ -522,7 +540,9 @@ SPECTRE_TEST_CASE(
   std::array<double, 5> second_order_error_6{};
   std::optional<double> dummy_expansion_velocity{};
   using DO = ::fd::DerivativeOrder;
-  for (const DO fd_do : {DO::Two, DO::Four, DO::Six, DO::Eight, DO::Ten}) {
+  for (const DO fd_do : {
+           DO::Two  // , DO::Four, DO::Six, DO::Eight, DO::Ten
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/Test_TimeDerivative.cpp:546]: add comment here why higher order is commented out

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TimeDerivative.hpp [hunk 4]
```diff
@@ -422,6 +446,17 @@ struct TimeDerivative {
           });
     }
 
+    if (UNLIKELY(fd_derivative_order != ::fd::DerivativeOrder::Two)) {
+      ERROR(
+          "We don't yet have high-order flux corrections for curved/moving "
+          "meshes and the implementation assumes curved/moving meshes. We need "
+          "to dot the Cartesian fluxes into the cell-centered "
+          "J inv(J)^{hat{i}}_j to get JF^{hat{i}} = J inv(J)^{hat{i}}_j F^j."
+          " Same care needs to be taken since we also get F^j from our "
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TimeDerivative.hpp]: some care

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TimeDerivative.hpp [hunk 5]
```diff
@@ -315,18 +319,35 @@ struct TimeDerivative {
...
+            //
+            // The unnormalized normal vector is
+            // n_j = d \xi^{\hat i}/dx^j
+            // with "i" the current face.
             tnsr::i<DataVector, 3, Frame::Inertial> lower_outward_conormal{
                 reconstructed_num_pts, 0.0};
-            lower_outward_conormal.get(i) = 1.0 / get(normalization);
+            for (size_t j = 0; j < 3; j++) {
+              lower_outward_conormal.get(j) =
+                  evolution::dg::subcell::fd::project_to_face(
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TimeDerivative.hpp]: rename project_to_faces (as I was confused about how what the target of the projection is)

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
Rebased and squashed. Thanks for the review!

**Review by kidder — APPROVED** [review]

**Pull request merged**